app/models/crypto_address_for_channel.rb

```diff
@@ -4,6 +4,16 @@ class CryptoAddressForChannel < ApplicationRecord
 
   SUPPORTED_CHAINS = ["ETH", "SOL"].freeze
   validates :chain, inclusion: {in: SUPPORTED_CHAINS}
-
   validates :channel_id, uniqueness: {scope: :chain}
+
+  scope :sol_addresses, -> { where(chain: "SOL") }
+  scope :eth_addresses, -> { where(chain: "ETH") }
+
+  def eth_address
+    eth_addresses.first.crypto_address.address
```

Comment:
What about the case of possible `nil` values for anything in the string here. Should we return `nil` or expect the data to be there? In general, if the data is expected to be there by the time this is called, we should blow up here and ensure data integrity in the caller/system. But if it's expected this will be called and might be `nil`, we'll want to use `&.` for the chain here.